app/grandchallenge/participants/form_fields.py

```diff
@@ -0,0 +1,36 @@
+from django.core.exceptions import ValidationError
+from django.forms import JSONField, TextInput
+from django.forms.fields import InvalidJSONInput
+from django.utils.html import escape
+
+
+class RegistrationQuestionField(JSONField):
+    empty_value = InvalidJSONInput("")
+
+    def __init__(self, *, registration_question, widget=TextInput, **kwargs):
+        self.registration_question = registration_question
+
+        kwargs.update(
+            {
+                "label": registration_question.question_text,
+                "required": registration_question.required,
+                "help_text": escape(registration_question.question_help_text),
+            }
+        )
+
+        if "initial" not in kwargs:
+            kwargs["initial"] = self.empty_value
+        else:
+            if isinstance(str, kwargs["initial"]):
+                kwargs["initial"] = InvalidJSONInput(kwargs["initial"])
+
+        super().__init__(widget=widget, **kwargs)
+
+    def to_python(self, value):
+        if value is None or value == "":
+            return self.empty_value
+        try:
+            return super().to_python(value)
+        except ValidationError:
+            # We assume it is a string
+            return value
```

Comment:
I don't really understand this block.

Reply:
Yeah, it had my head in a twist for a bit as well. Most (all?) input widgets produce a pure string value.

If one would want to input the empty string you'd have to type:
`""` into the HTML input. A regular string would be `"Hello world"` instead of `Hello world`.

That is:
`json.loads("Hello world")` vs `json.loads("\"Hello world\"")`.

This block allows the field to interpret `Hello world` as a valid string input.

Reply:
What ValidationError would we expect to catch here? Is there a code attached to it? Should we check that the value is a string rather than assuming it?

Reply:
Looking at the problem at hand closer, I also added some tests which had me in turn make the to_python more complex to handle double-quotation marks in different numbers, locations etc.

Reply:
Resolved, I think. Leaving it up to you James.